servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -345,8 +351,11 @@ public void commitBlock(long workerId, long usedBytesOnTier, String tierAlias, l
         if (masterBlockInfo == null) {
           masterBlockInfo = new MasterBlockInfo(blockId, length);
           mBlocks.put(blockId, masterBlockInfo);
-          writeJournalEntry(
-              new BlockInfoEntry(masterBlockInfo.getBlockId(), masterBlockInfo.getLength()));
+          BlockInfoEntry blockInfo = BlockInfoEntry.newBuilder()
+              .setBlockId(masterBlockInfo.getBlockId())
+              .setLength(masterBlockInfo.getLength())
+              .build();
+          writeJournalEntry(JournalEntry.newBuilder().setBlockInfo(blockInfo).build());
```

Comment:
The overloading approach adds a lot of boilerplate with all the overloaded methods, and the `unwrap` requires passing the journal entry to be wrapped as the `Message` type, then using a long chain of `instanceof` to recover the right type. I'm also quite bothered by the boilerplate in wrapping the entries, but the alternatives don't seem much better. Anyway, this PR is getting pretty long, so lets revisit this issue later